libphal/phal_pdbg.C

```diff
@@ -280,4 +280,31 @@ bool isSbeVitalAttnActive(struct pdbg_target *proc)
 	return validAttn;
 }
 
+bool hasControlTransitionedToHost()
+{
+  // If unable to read the register, by default consider the control is in hostboot
+  auto transitionedToHost = false;
+  //Read the scratch register to find if the control has moved to phyp (Host)
+  auto pibTarget = pdbg_target_from_path(nullptr, "/proc0/pib");
+
+  uint64_t l_coreScratchRegData = 0xFFFFFFFFFFFFFFFFull;
+  // HB changes the below core scratch reg as one of the last instructions that is run,
+  // so if that is zero then we're in phyp
+  uint64_t l_coreScratchRegAddr = 0x4602F489;
+
+  // Is there any error in reading the scom address, consider control is in hostboot
+  if (pib_read(pibTarget, l_coreScratchRegAddr, &l_coreScratchRegData))
+  {
+      log(level::ERROR, "scom read error: 0x%X", l_coreScratchRegAddr );
+      return false;
+  }
+
+  // If the register reads zero, return control moved to phyp.
+  if(l_coreScratchRegData == 0)
```

Comment:
we can directly do a `return (l_coreScratchRegData == 0);` here and can eliminate the need of `transitionedToHost` variable declared at LN286. Just a suggestion.

Reply:
Thank you. that makes it simple. Will address